boards/arm/tdk_robokit1/zephyr_thermistor.dtsi

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2023 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* NTC Thermistor Table Generated with ntc_thermistor_table.py */
+
+/ {
+	thermistor_R25_10000_B_3974: thermistor-R25-10000-B-3974 {
+		status = "disabled";
+		compatible = "zephyr,thermistor-rt-table";
```

Comment:
I wanted to show some association between the temperature and resistance.  Having individual arrays for temperature and another for resistance you lose that association in the dts.  

I tried using a single array (see below) but can't seem to figure out a way to `index+=2` when initializing.  Any suggestions with this method?

```
		/* Format <temp resistance> */
		tr-table = <(-25) 146676>,
		           <(-15) 78875>,
		           <(-5) 44424>,
		           <(5) 26075>,
		           <(15) 15881>,
		           <(25) 10000>,
		           <(35) 6488>,
		           <(45) 4326>,
		           <(55) 2956>,
		           <(65) 2066>,
		           <(75) 1474>,
		           <(85) 1072>,
		           <(95) 793>,
		           <(105) 596>,
		           <(115) 454>,
		           <(125) 351>;
```

Sticking with the child node approach, I can clean-up the redundant `num-pairs` and `sample-index` properties.

Reply:
Are they both encoded as int32's in dt? If that's case, could just store them as a 1d int32 array and then cast a pointer.

I'll see if I can find the right place to comment on code..

Reply:
You'll have to create a single array like:
```
		/* Format <temp resistance> */
		tr-table = <(-25) 146676
		           (-15)  78875
		           (-5)   44424
		           (5)    26075
		           (15)   15881
		           (25)   10000
		           (35)   6488
		           (45)   4326
		           (55)   2956
		           (65)   2066
		           (75)   1474
		           (85)   1072
		           (95)   793
		           (105)  596
		           (115)  454
		           (125)  351>;
```

Reply:
I updated `struct ntc_compensation` to be both 32 bit values.  I then cast the uint32_t array to a `struct ntc_compensation *`  this lets me keep the association in the device tree.  I attempted to create a union of both pointer types to avoid casting but compiler kept giving `error: initializer element is not constant`.